### ISSUE TITLE
[Feature] Linux/Steam Deck Support

### ISSUE THREAD

**Issue by bkapvt** [vs-shirokii/gzdoom-rt#26]
### Which game are you running with GZDoom?

None

### What Operating System are you using?

None

### Is your feature request related to a problem? Please describe.

Not supported on linux it seems

### Describe the solution you'd like

Linux! Steam Deck! Would deck even be able to run this thing?

**Comment by KingVulpes** [comment]
Wondering the same thing. It won't even open on SteamOS

I do see that other Linux distros are having better luck so SteamOS might just have it's own issues

**Comment by Vash63** [comment]
I reported this for Proton also, it looks like this relies on a Vulkan extension that the latest Proton doesn't translate: `VK_KHR_external_fence_win32`
https://github.com/ValveSoftware/Proton/issues/1534#issuecomment-2295254691

**Comment by KingVulpes** [comment]
> I reported this for Proton also, it looks like this relies on a Vulkan extension that the latest Proton doesn't translate: `VK_KHR_external_fence_win32` [ValveSoftware/Proton#1534 (comment)](https://github.com/ValveSoftware/Proton/issues/1534#issuecomment-2295254691)

It can be switched to DXGI, tho I haven't managed to find the config file location on Steam Deck yet

**Comment by ryanmusante** [comment]
So it is not currently compatible with Linux in this version?

**Comment by Vash63** [comment]
> So it is not currently compatible with Linux in this version?

Correct. Proton translates some `_win32` extensions in the winevulkan component, but `VK_KHR_external_fence_win32` doesn't seem to be translated yet. It might be easier to just make a native Linux port given gzdoom already supports it, but would still need someone who knows Vulkan to make sure any platform specific new things added in this fork are updated for Linux.

**Comment by KingVulpes** [comment]
It seems SteamOS is more broken than other distros, because nothing loads on Deck, but it seems other distros at least get to the loader

**Comment by Vantskruv** [comment]
I have been trying to compile it on Linux in hope to get it to run, but have not come so far I think. Note that I am far from an expert when it comes to cmake and its arguments.

I have been trying to going through the steps in _auto-setup-windows.cmd_ file. I got some x64 errors when running cmake. So I removed the -A x64 arguments in the command lines in the file, also I removed the "-- -maxcpucount -verbosity:minimal" argument in the command when building.

I was successfull with these two commands (linking I guess, and compiling) zmusic:
```
cmake -S ./zmusic -B ./zmusic/build ^
	-DCMAKE_TOOLCHAIN_FILE=../vcpkg/scripts/buildsystems/vcpkg.cmake ^
	-DVCPKG_LIBSNDFILE=1 ^
	-DVCPKG_INSTALLLED_DIR=../vcpkg_installed/
cmake --build ./zmusic/build --config Release
```

Sadly the first command (see below) fails in the second pair of commands, with an x64 error, even if removing the -A x64 argument from the command.
Notice that I changed the DZMUSIC_LIBRARIES from zmusic.lib to libzmusic.so (no Release folder has been created, I do not understand why, but the linux library files has been compiled and created in the source directory).
This is the command I used which fails:
```
cmake -S .. -B . ^
	-DCMAKE_TOOLCHAIN_FILE=./vcpkg/scripts/buildsystems/vcpkg.cmake ^
	-DZMUSIC_INCLUDE_DIR=./zmusic/include ^
	-DZMUSIC_LIBRARIES=./zmusic/build/source/libzmusic.so ^
	-DVCPKG_INSTALLLED_DIR=./vcpkg_installed/

```
and I get this output ( I only copied the output near the error) :
```
...

The package bzip2 is compatible with built-in CMake targets:

    find_package(BZip2 REQUIRED)
    target_link_libraries(main PRIVATE BZip2::BZip2)

libvpx provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(unofficial-libvpx CONFIG REQUIRED)
  target_link_libraries(main PRIVATE unofficial::libvpx::libvpx)

libvpx provides pkg-config modules:

  # WebM Project VPx codec implementation
  vpx

libwebp provides CMake targets:

    find_package(WebP CONFIG REQUIRED)
    # basic usage
    target_link_libraries(main PRIVATE WebP::webp WebP::webpdecoder WebP::webpdemux)
    # for manipulating the WebP graphics format container (port feature libwebpmux)
    target_link_libraries(main PRIVATE WebP::libwebpmux)
    # for sharp RGB to YUV conversion
    target_link_libraries(main PRIVATE WebP::sharpyuv)

sdl2 provides CMake targets:

    find_package(SDL2 CONFIG REQUIRED)
    target_link_libraries(main
        PRIVATE
        $<TARGET_NAME_IF_EXISTS:SDL2::SDL2main>
        $<IF:$<TARGET_EXISTS:SDL2::SDL2>,SDL2::SDL2,SDL2::SDL2-static>
    )

-- Running vcpkg install - done
CMake Error at CMakeLists.txt:32 (project):
  Generator

    Unix Makefiles

  does not support platform specification, but platform

    x64

  was specified.


CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!

```
I am not sure which CMakeLists.txt file gives this error, there is hundreds of CMakeLists.txt files.
So currently I am stuck.
I am not sure if anyone else have come longer with the tries of compiling gzdoom-rt for Linux.

**Comment by Vantskruv** [comment]
I came a little further now, I removed the extra letter 'L' in the argument DVCPKG_INSTALLLED_DIR in all the commands, so the argument is not DVCPKG_INSTALLED_DIR, correctly spelled now.

The first command in the second pair of commands now works:
```
cmake -S .. -B . ^
	-DCMAKE_TOOLCHAIN_FILE=./vcpkg/scripts/buildsystems/vcpkg.cmake ^
	-DZMUSIC_INCLUDE_DIR=./zmusic/include ^
	-DZMUSIC_LIBRARIES=./zmusic/build/source/libzmusic.so ^
	-DVCPKG_INSTALLED_DIR=./vcpkg_installed/
```
But guess what, the second and last command to compile the code fails at the start.
`$ cmake --build . --config RelWithDebInfo`
So I am now trying to solve this error output when typing the above command:

```
[  0%] Building C object libraries/lzma/CMakeFiles/lzma.dir/C/7zAlloc.c.o
cc: warning: /DHAVE_RT=0: linker input file unused because linking not done
cc: error: /DHAVE_RT=0: linker input file not found: No such file or directory
make[2]: *** [libraries/lzma/CMakeFiles/lzma.dir/build.make:76: libraries/lzma/CMakeFiles/lzma.dir/C/7zAlloc.c.o] Error 1
make[2]: *** Deleting file 'libraries/lzma/CMakeFiles/lzma.dir/C/7zAlloc.c.o'
make[1]: *** [CMakeFiles/Makefile2:352: libraries/lzma/CMakeFiles/lzma.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

```

**Comment by this-is-just-con-tribute** [comment]
> I reported this for Proton also, it looks like this relies on a Vulkan extension that the latest Proton doesn't translate: `VK_KHR_external_fence_win32` [ValveSoftware/Proton#1534 (comment)](https://github.com/ValveSoftware/Proton/issues/1534#issuecomment-2295254691)

any update on this?

**Comment by DevDema** [comment]
I don't see why this was not developed. GZdoom has active support for both macOS and Linux, and yet only the windows version was modded?

**Comment by Vash63** [comment]
> I don't see why this was not developed. GZdoom has active support for both macOS and Linux, and yet only the windows version was modded?

Because the author isn't a Linux user? I agree that it's a shame but an open source author really has no obligation to develop for platforms or people other than themselves. (btw, has anyone tried with Proton 11 yet? My initial tests here were years ago and apparently Proton devs have worked on these extensions since)
